### PR TITLE
Updated conditional rendering doc with a note on Falsy values

### PR DESCRIPTION
**Pull request by jeswinsimon**
Conditional rendering works if the expression returns a truthy/falsy value instead of true/false but the falsy value ends up being rendered. This can cause an unexpected value being rendered on screen. 

The discussion regarding this can be found [here](https://github.com/facebook/react-native/issues/29980#issuecomment-696437692)

### CODE REVIEW

File: content/docs/conditional-rendering.md [hunk 1]
```diff
@@ -152,6 +152,19 @@ It works because in JavaScript, `true && expression` always evaluates to `expres
 
 Therefore, if the condition is `true`, the element right after `&&` will appear in the output. If it is `false`, React will ignore and skip it.
 
+Note that returning a falsy expression will still cause the element after `&&` to be skipped but will return the falsy expression. In the example below, 0 will be returned by the expression.
```
Comment (eps1lon) [inline content/docs/conditional-rendering.md:155]: I think we can be a bit more specific what "the expression" is and instead explain directly what is returned from render. `<div>0</div>`

**Review by eps1lon — CHANGES_REQUESTED** [review]
Explaining the drawback of shorthand-conditional rendering is indeed important, thanks! Just a suggestion on how the communicate the implications.

File: content/docs/conditional-rendering.md (hunk 1 above)
Comment (jeswinsimon) [reply content/docs/conditional-rendering.md]: Thanks! That makes more sense.

**Review by eps1lon — APPROVED** [review]

**Comment by eps1lon** [comment]
@jeswinsimon Thanks!

**Pull request merged**